src/main/scala/chisel3/util/circt/PlusArgsValue.scala

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.circt
+
+import chisel3._
+import chisel3.experimental.{FlatIO, IntrinsicModule}
+import chisel3.internal.Builder
+
+import circt.Intrinsic
+
+/** Create a module which generates a verilog \$plusargs\$value.  This returns a
+  * value as indicated by the format string and a flag for whether the value
+  * was found.
+  */
+private class PlusArgsValueIntrinsic[T <: Data](gen: T, str: String)
+    extends IntrinsicModule("circt.plusargs.value", Map("FORMAT" -> str)) {
+  val io = FlatIO(new Bundle {
+    val found = Output(UInt(1.W))
+    val result = Output(gen)
+  })
```

Comment:
You can directly use `val found = Output(UInt(1.W))` instead of passing this through `FlatIO`. That really only becomes necessary when dealing with a `BlackBox` which dynamically checks that there is a `val io`.

Same nit as above, `Bool` is likely better for `found`.

For `result`, is this appropriate to take a type parameter? It seems like we wouldn't know what to do with this if it was anything other than a `UInt`? Would this be better expressed as a `UInt` of fixed width (for now) or parametric width (later)?

Reply:
I think @seldridge meant `val found = IO(Output(Bool())` (edit mine on the Bool, but the point is you do still need the `IO`, but you're allowed to have as many `IO` calls as you want.

Reply:
and we could still take a gen: UInt, and the user controls the width. Alternatively the old plus arg reader just forced you to specify the width and then it made the UInt with that width, no gen.

Reply:
FlatIO was an old artifact of extmodule.  Fixed.

Reply:
I take that back, this gives me a bundle for scala purposes but  flat io in the module.

Reply:
I care about exposing verilog functionality via an intrinsic, not about doing exactly what one code bases blackbox does.